### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by Movgto**
### **In this branch we have worked on the following tasks :** 

- On exit the app saves all the data (books, people, and rentals) into JSON files. One per type of data is stored and named accordingly.
- On startup, the app loads all data from files. Handling the case when one or more of the files are missing.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -15,48 +17,40 @@ def option_list
...
+
+    data = {
+      people: people,
+      books: books,
+      rentals: rentals
+    }
+
+    options = {
+      '1' => method(:get_list_books),
+      '2' => method(:get_list_person),
```
Comment (Olamarx) [inline main.rb]: - [ ] This throws error because you actually have get_list_people, not person. Kindly fix it up.

File: main.rb [hunk 2]
```diff
@@ -15,48 +17,40 @@ def option_list
...
+      '3' => method(:add_new_person),
+      '4' => method(:add_new_book),
+      '5' => method(:create_new_rental),
+      '6' => method(:get_user_rental),
```
Comment (Olamarx) [inline main.rb]: - [ ] I could not find the get_user_rental method and it throws error, I will encourage you to create it.

File: main.rb [hunk 3]
```diff
@@ -15,48 +17,40 @@ def option_list
...
+      '7' => method(:exit)
```
Comment (Olamarx) [inline main.rb]: - [ ] The exit throws an error though the data saves but it should exit gracefully without error, kindly double-check your code.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi Team,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Commit messages meaningful ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — APPROVED** [review]
# Hi @Movgto & @ZabihullahNooriWardak 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Data is properly preserved in JSON files ✔
- Data is properly retrieved from the files ✔
- Case where file doesn't exist is properly handled ✔
- Correct use of Git Flow ✔
- Linter check is passing ✔
- Descriptive PR title with a good summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**